src/chunk-stream.iced

```diff
@@ -8,40 +9,62 @@ exports.ChunkStream = class ChunkStream extends stream.Transform
 
   _transform : (chunk, encoding, cb) ->
     # if we're processing objects, just go one at a time, ignoring everything else
-    if @writableObjectMode
-      @push(@transform_func(chunk))
-      return cb()
+    esc = make_esc(cb, "Error in transform function")
 
-    # prepend any extra
-    if @extra?
-      chunk = Buffer.concat([@extra, chunk])
-      @extra = null
+    if @writableObjectMode
+      await @transform_func(chunk, esc(defer(out)))
+      @push(out)
+      return cb(null)
 
-    # if we don't have a full block, dump everything into extra and skip this round
-    if chunk.length < @block_size
-      @extra = chunk
     else
-      # if we have to, call the transform function multiple times until we have less than a block size remaining
-      if @exact_chunking
-        while chunk.length >= @block_size
-          @push(@transform_func(chunk[...@block_size]))
-          chunk = chunk[@block_size...]
+      # prepend any extra
+      if @extra?
+        chunk = Buffer.concat([@extra, chunk])
+        @extra = null
+
+      # if we don't have a full block, dump everything into extra and skip this round
+      if chunk.length < @block_size
         @extra = chunk
-      # if the transform function can accept chunks of length block_size*n for n \in N, just make one transform function call
       else
-        remainder = chunk.length % @block_size
-        # chop off extra
-        if remainder isnt 0
-          @extra = chunk[chunk.length-remainder...]
-          chunk = chunk[...chunk.length-remainder]
-        @push(@transform_func(chunk))
+        # if we have to, call the transform function multiple times until we have less than a block size remaining
+        if @exact_chunking
+          while chunk.length >= @block_size
+            await @transform_func(chunk, esc(defer(out)))
+            @push(out)
+            chunk = chunk[@block_size...]
+          @extra = chunk
+          return cb(null)
```

Comment:
Ditto. One of the goals of early exiting is to avoid the deep levels of indentation.